fendermint/vm/interpreter/src/fvm/gas/actor.rs

```diff
@@ -17,9 +18,21 @@ pub struct ActorGasMarket {
     block_gas_limit: Gas,
     /// The accumulated gas usage so far
     block_gas_used: Gas,
+    /// Pending update to the underlying gas actor
+    constant_update: Option<SetConstants>,
 }
 
 impl GasMarket for ActorGasMarket {
+    type Constant = SetConstants;
+
+    fn get_constants(&self) -> anyhow::Result<Self::Constant> {
+        todo!()
+    }
+
+    fn set_constants(&mut self, constants: Self::Constant) {
+        self.constant_update = Some(constants);
+    }
+
```

Comment:
Are `GetConstants` and `SetConstants` ever used by the App? These are policy-specific methods for the user to invoke off band.

Reply:
Yep, they are not used by the `App` and called by system upgrade, maybe.

I added these two for "reference" implementation I guess. The `SetConstants` is also related to how `state` could detect potential gas market changes. I'm thinking of holding the changes in memory first, as reflected in the option field, then in `end_block` performs the commit in interpreter. Then send a summary of the changed to `App` which then can be propagated to cometbft. If we dont control `SetConstants`, we might lost the change or have to query the gas market over and over to detect changes.